pkg/operator/webhooksupportabilitycontroller/degraded_webhook.go

```diff
@@ -101,8 +102,15 @@ func (c *webhookSupportabilityController) assertConnect(ctx context.Context, ref
 		port = fmt.Sprintf("%d", *reference.Port)
 	}
 	rootCAs := x509.NewCertPool()
-	if len(caBundle) > 0 {
+	switch {
+	case net.JoinHostPort(host, port) == "kubernetes.default.svc:443":
+		// ignore any caBundle specified and get one from the service account
+		rootCAs = serviceAccountCABundle()
+	case len(caBundle) > 0:
 		rootCAs.AppendCertsFromPEM(caBundle)
+	default:
+		// if no caBundle was specified, use the one from the service account
+		rootCAs = serviceAccountCABundle()
```

Comment:
talked with @deads2k in slack

> if the host != "kubernetes.default.svc", then serviceaccount/ca.crt should not be used by default.

it seems we should not default it here , then?